src/pds/peppi/client.py

```diff
@@ -419,3 +432,43 @@ def reset(self):
         self._expected_pages = None
         self._page_counter = None
         self._latest_harvest_time = None
+
+    def as_dataframe(self, max_rows: Optional[int] = None):
```

Comment:
Does it might make more sense to define this as a static method on the `Products` class, rather than as an instance method?

The calling convention could then be:
```python
# Set up query
products.has_target("...").has_instrument("...")

# Obtain results via iteration
for p in products:
     <do something>

# Obtain results as a dataframe
df = Products.as_dataframe(products, max_rows = 10)
```

My concern is that adding a `as_dataframe` call to the end of a chain of query setup calls will be confusing to an end-user, and sort of interferes with the "lazy evaluation" paradigm used to set up queries.

Reply:
@collinss-jpl I've got your point, especially that makes the `h` or `has_target` or `has_instrument` very important 😄 .  

I started to implement your proposal but had an issue with the static method using the class as an argument type:

`def to_dataframe(product: Products, max_rows...):`

It is some kind of cyclic dependency.

I will take some more time to think of it or find examples elsewhere since I am not totally convince we need to go through this static method.

Reply:
@tloubrieu-jpl I believe the cyclic dependency is only caused by including the type hint for the `product: Products` argument. I was able to it working by just omitting the type-hint. The following code sample shows the changes I had to make on my local copy, including updates to the unit tests:

https://gist.github.com/collinss-jpl/9eb1c1c59cc52207f3161607898aa6a9

I would be happy to discuss further at today's breakout, but I think I have a conflict for that time.

Reply:
The topic has been discussed at the breakout and on slack with @nutjob4life, @collinss-jpl and @jordanpadams , we keep going with the as_dataframe carried by the same object Product object as the filter methods.
However the code is distributed under the hood in different classes.